core/src/main/java/jenkins/util/AntClassLoader.java

```diff
@@ -793,21 +816,7 @@ private InputStream getResourceStream(File file, String resourceName) {
                     return Files.newInputStream(resource.toPath());
                 }
             } else {
-                if (jarFile == null) {
-                    if (file.exists()) {
-                        jarFile = new JarFile(file);
-                        jarFiles.put(file, jarFile);
-                    } else {
-                        return null;
-                    }
-                    //to eliminate a race condition, retrieve the entry
-                    //that is in the hash table under that filename
-                    jarFile = (JarFile) jarFiles.get(file);
-                }
-                JarEntry entry = jarFile.getJarEntry(resourceName);
-                if (entry != null) {
-                    return jarFile.getInputStream(entry);
-                }
+                return storeAndConvert<InputStream>(jarFile, file, resourceName, (jar, entry) -> jar.getInputStream(entry));
```

Comment:
The syntax is incorrect here. It should be 

```suggestion
                return storeAndConvert(jarFile, file, resourceName, (jar, entry) -> jar.getInputStream(entry));
```

or even

```suggestion
                return storeAndConvert(jarFile, file, resourceName, JarFile::getInputStream);
```

Reply:
This appears to be undoing the suggestion made by @oleg-nenashev which I applied as 6485ce053cb6d967fc9347846cb9af3aa1f2ab5e

Reply:
Actually not, Oleg already proposed what Adrien proposes here :) it's the difference between explicit casting and generic usage. So that's fine after the modifications.